### PR TITLE
[docs] Migrate the Activity module docs

### LINKED ISSUE #21: [docs] Migrate the "Activity modules"
Migrate https://docs.moodle.org/dev/Activity_modules

Proposed to: `/docs/plugintypes/mod` but possibly under `/docs/apis/plugintypes/mod` ?

### PR DESCRIPTION
**Pull request by andrewnicols**
This commit includes changes to remove lots of out-dated code, and to
point to other documentation where possible.

Fixes #21

### CODE REVIEW

File: docs/apis/plugintypes/mod.md [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        $mform->addHelpButton('usecode', 'usecode', 'certificate');
+
+        $this->standard_coursemodule_elements();
+
+        $this->add_action_buttons();
+    }
+}
+```
+
+
```
Comment (sarjona) [inline docs/apis/plugintypes/mod.md:379]: Lint has raised this consecutive blank line

File: docs/apis/plugintypes/mod.md [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+$instance = $DB->get_record('[modname]', ['id'=> $cm->instance], '*', MUST_EXIST);
+```
+
+### `version.php`
+
+The `version.php` file keeps track of the version of your module, and other attributes, such as what version of Moodle it requires. For a full list of the attributes please see [[version.php]].
+
+## See also
+
+* [[NEWMODULE_Documentation]]
```
Comment (sarjona) [inline docs/apis/plugintypes/mod.md:426]: Lint has raised "Unordered list style [Expected: dash; Actual: asterisk]"

**Review by sarjona — CHANGES_REQUESTED** [review]
The Activity modules doc looks great (and I've also learnt the `_category_.yml` file). As you'll see, there are a couple of tiny things to fix before merging it :-)
Thanks Andrew!! :-*

**Comment by andrewnicols** [comment]
Thanks - fixed those two stupid mistakes.

I'm going to make linting happen as a pre-commit hook!

**Review by sarjona — APPROVED** [review]
Thanks Andrew!
To make linting happen as a pre-commit is a good idea!! <3

**Pull request merged**